### DISCUSSION TITLE
Define Custom Tax Provider

### DISCUSSION THREAD

**Discussion by ricardolopesdev** [medusajs/medusa#10140] [Feature Requests]
It's any way to asign a custom Tax Provider to current Region?
At moment the only way so far to define the tax provider in the region is to directly change the DB.
First in table "**_tax_provider_**" i added the **id** of my provider in this case "**my-tax**" and change the flag to **is_enable** to "**true**", after that it's needed go to table "tax_region" and asign the _id of provider_ to field "**prodiver_id**".
Thanks

**Comment by peakears** [comment]
I'm looking for a way to do this programatically? Any update on this feature?

I would like to set a custom tax provider to be used by specific tax regions. Currently, I can only do this manually by editing the tables by hand.

**Comment by fPolic** [comment]
Hey guys, we are finalising the implementation for supporting custom tax providers, so hopefully it will be available in the next release.

**Reply by fxmb** [reply]
Is there a way you can parse the current cart object in the getTaxLines() context as well? Use case is that in Europe there can arise tricky situations with taxes based on the customers delivery and shipping country especially in B2B use cases. Almost impossible to solve wihout the freedom of coding access.

**Reply by fxmb** [reply]
Currently only the shipping address is parsed... which is sometimes not enough to calculate the correct amount of tax

**Comment by metteokiehn-collab** [comment]
The ability to define a custom tax provider offers greater flexibility when managing property-related tax information. It can help users organize data sources according to specific requirements and workflows. For anyone researching assessments and ownership details https://indiancountypropertyappraiser.org can be a useful resource for exploring relevant information. This enhancement makes tax configuration more adaptable and convenient for different use cases.
